images/router/haproxy/conf/haproxy-config.template

```diff
@@ -769,6 +769,23 @@ backend {{ genBackendNamePrefix $cfg.TLSTermination }}:{{ $cfgIdx }}
               {{- else }}
                 {{- if not (isTrue $router_disable_http2) }} alpn h2,http/1.1
                 {{- end }}
+                {{- /*
+                    Add "verifyhost" with service hostname only if both are true:
+                    - The route has a single active service.
+                    - The route does not have a destination CA certificate.
+
+                    Limiting "verifyhost" to a single service route is essential,
+                    as it's not possible to reliably predict which service’s endpoint will be used
+                    when alternate backends are available.
+                */ -}}
+                {{- if and $cfg.VerifyServiceHostname (eq $cfg.ActiveServiceUnits 1) -}}
```

Comment:
So if `cfg.ActiveServiceUnits > 1`, then this entire reencrypt backend is effectively not dynamic (that's unfortunate). However, the config still shows up with the dynamic servers:
```
  server _dynamic-pod-1 172.4.0.4:8765 weight 0 ssl disabled check inter 5000ms verify required ca-file /var/run/configmaps/service-ca/service-ca.crt
  server _dynamic-pod-2 172.4.0.4:8765 weight 0 ssl disabled check inter 5000ms verify required ca-file /var/run/configmaps/service-ca/service-ca.crt
  server _dynamic-pod-3 172.4.0.4:8765 weight 0 ssl disabled check inter 5000ms verify required ca-file /var/run/configmaps/service-ca/service-ca.crt
```

Did you consider moving this condition up so that we don't even print any dynamic server lines? They are totally unusable right? I know it's mostly a NO-OP, but it might be more explicit when debugging the config to not mislead folks into thinking this dynamic servers are actually useful.

EDIT: See my next comment since this might be OBE.